projects/cashmere/src/lib/select/select.component.ts

```diff
@@ -34,17 +38,14 @@ export class SelectChangeEvent {
     encapsulation: ViewEncapsulation.None,
     providers: [{provide: HcFormControlComponent, useExisting: forwardRef(() => SelectComponent)}]
 })
-export class SelectComponent extends HcFormControlComponent implements ControlValueAccessor, AfterContentInit, DoCheck {
+export class SelectComponent extends HcFormControlComponent implements ControlValueAccessor, DoCheck {
     private _uniqueInputId = `hc-select-${uniqueId++}`;
     private _form: NgForm | FormGroupDirective | null;
     private _tight: boolean = false;
     private _value: any = '';
-    private _valueData: any;
-
-    _componentId = this._uniqueInputId;
-
-    @ContentChildren(HcOptionDirective)
```

Comment:
Following angular's lead and having select options register themselves with the select component, rather than querying for them from the parent.